src/objclass/class_api.cc

```diff
@@ -66,6 +66,15 @@ int cls_register_cxx_method(cls_handle_t hclass, const char *method,
   return (hmethod != NULL);
 }
 
+// deprecated
+int cls_register_cxx_method(cls_handle_t hclass, const char *method,
```

Comment:
We'll want eventually to switch all our _plugins_ into the new interface. If external ones can be neglected (I would argue for that), the entire deprecated registration machinery can be dropped.